pkg/collector/edp/collector.go

```diff
@@ -2,69 +2,192 @@ package edp
 
 import (
 	"context"
-	"maps"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
 	"time"
 
 	"go.opentelemetry.io/otel"
-	"go.opentelemetry.io/otel/attribute"
-	"go.opentelemetry.io/otel/trace"
-	"go.uber.org/zap"
+	"go.opentelemetry.io/otel/codes"
 
 	"github.com/kyma-project/kyma-metrics-collector/pkg/collector"
+	kmcotel "github.com/kyma-project/kyma-metrics-collector/pkg/otel"
 	"github.com/kyma-project/kyma-metrics-collector/pkg/resource"
 	"github.com/kyma-project/kyma-metrics-collector/pkg/runtime"
-	skrcommons "github.com/kyma-project/kyma-metrics-collector/pkg/skr/commons"
 )
 
 type Collector struct {
-	scanners []resource.Scanner
-	logger   *zap.Logger
+	EDPClient *Client
+	scanners  []resource.Scanner
 }
 
+var errNoMeasurementsSent = errors.New("no measurements sent to EDP")
+
 var _ collector.CollectorSender = &Collector{}
 
-func NewCollector(scanner ...resource.Scanner) collector.CollectorSender {
+func NewCollector(EDPClient *Client, scanner ...resource.Scanner) collector.CollectorSender {
 	return &Collector{
-		scanners: scanner,
+		EDPClient: EDPClient,
+		scanners:  scanner,
 	}
 }
 
 func (c *Collector) CollectAndSend(ctx context.Context, runtime *runtime.Info, previousScans collector.ScanMap) (collector.ScanMap, error) {
-	childCtx, span := otel.Tracer("").Start(ctx, "collect",
-		trace.WithAttributes(
-			attribute.String("provider", runtime.ProviderType),
-			attribute.String("shoot_id", runtime.ShootInfo.ShootName),
-		),
-	)
+	var errs []error
+
+	childCtx, span := otel.Tracer("").Start(ctx, "kmc.collect_scans_and_send_measurements", kmcotel.SpanAttributes(runtime))
```

Comment:
```suggestion
	childCtx, span := otel.Tracer("").Start(ctx, "collect_scans_and_send_measurements", kmcotel.SpanAttributes(runtime))
```

"kmc" should refer to the service name. The service name can be set using environment variables in the helm chart